python_modules/dagster/dagster/_core/ext/subprocess.py

```diff
@@ -88,6 +89,9 @@ def run(
                 raise DagsterExternalExecutionError(
                     f"External execution process failed with code {process.returncode}"
                 )
+            _ext_context = ext_context
```

Comment:
did you get some error that caused to you do this re-assign? From what I understand the values from a `with` statement are not scoped to the opened block, you have access to them post exit. 

I think we can just do `ext_context.get_materialize_results()` and have some explicit runtime error if its called too early. Might need to have `ExtOrchestrationContext(ContextManager)` to do it cleanly

Reply:
>did you get some error that caused to you do this re-assign? From what I understand the values from a with statement are not scoped to the opened block, you have access to them post exit.

Hmm you're right. Not sure why I thought the yielded var became unbound.

>I think we can just do ext_context.get_materialize_results() and have some explicit runtime error if its called too early. Might need to have ExtOrchestrationContext(ContextManager) to do it cleanly

I will do a quick rejigger

Reply:
Did the rejigger. I just made `ext_protocol` set an `is_task_finished` on `ExtOrchestrationContext` when it exits. You are correct that we just care about the message reader, and we know the message reader has exited if `ext_protocol` has.